definitions/procedures/restore/ensure_mongo_engine_matches.rb

```diff
@@ -0,0 +1,28 @@
+module Procedures::Restore
+  class EnsureMongoEngineMatches < ForemanMaintain::Procedure
+    metadata do
+      description 'Ensure restored MongoDB storage engine matches the current DB'
+      for_feature :mongo
+    end
+
+    def run
+      if feature(:mongo).local? && engine_mismatch?
+        with_spinner('') do |spinner|
+          feature(:service).handle_services(spinner, 'stop', :only => feature(:mongo).services)
+          spinner.update('Clean MongoDB data')
+          data_path = Dir[feature(:mongo).data_dir + '/*']
+          FileUtils.rm_rf(data_path)
```

Comment:
I'm always a bit cautions when I see `rm_rf` somewhere. I would rather double-check, that the `feature(:mongo).data_dir` doesn't return `nil?` or `empty?` so that we don't end up with `rm_rf("/*")`, as for example https://github.com/theforeman/foreman_maintain/blob/3da185523aba488b4118a991ca783e609429f583/definitions/procedures/backup/offline/mongo.rb#L27 has some corner cases.